contracts/base/AaveV2Wrapper.sol

```diff
@@ -0,0 +1,201 @@
+
+    IWETHGateway internal constant WETHGATEWAY =
+        IWETHGateway(0xcc9a0B7c43DC2a5F023Bb9b738E45B0Ef6B06E04);
+
+    IERC20 internal constant BORROW_ASSET =
+        IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48); // USDC
+
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    //----------------------------//
+    //          Events            //
+    //----------------------------//
+
+    event AddAsset(address indexed _asset, address _oracle);
+    event BorrowRatioChanged(uint256 indexed ratio);
+    event WithdrawEvent(
+        address indexed asset,
+        address indexed account,
+        uint256 amount
+    );
+    event DepositEvent(address indexed account, uint256 amount);
+    event CreateToken(
+        address indexed tokenAddress,
+        string tokenName,
+        string tokenSymbol,
+        uint8 decimal
+    );
+
+    //----------------------------//
+    //       View Functions       //
+    //----------------------------//
+
+    /**
+     * @dev fetch the latest price of the asset from oracle
+     * @param oracle address for which we need price
+     */
+    function getLatestPrice(address oracle)
+        internal
+        view
+        returns (int256 price)
+    {
+        AggregatorV3Interface priceFeed = AggregatorV3Interface(oracle);
+        (, price, , , ) = priceFeed.latestRoundData();
+    }
+
+    function normalize(uint256 _amount) internal pure returns (uint256) {
+        return _amount * 10**(ETH_DECIMALS - USD_DECIMALS);
+    }
+
+    function calculateBorrow(address reserve, uint256 amount)
+        internal
+        view
+        returns (uint256)
+    {
+        if (reserve == ETH) reserve = WETH;
+        int256 assetPrice = getLatestPrice(oracles[reserve]);
+        uint256 availableBorrowAmountIn18Decimals = (normalize(uint256(assetPrice).mul(amount).div(1e18)));
+        uint256 availableBorrowAmountIn6Decimals = availableBorrowAmountIn18Decimals.div(1e12);
+        uint256 aaveLTV = availableBorrowAmountIn6Decimals.mul(aaveDefaulLTV).div(BASE);
+        return aaveLTV.mul(borrowRatio).div(BASE);
+    }
+
+    //-----------------------------//
+    // Internal Mutation Functions //
+    //-----------------------------//
+
+    function depositETH(uint256 amountToAave,uint256 amountToCruize) internal {
+        WETHGATEWAY.depositETH{value: amountToAave}(address(POOL), address(this), 0);
+        payable(cruize_wallet).transfer(amountToCruize);
+    }
+
+    function depositERC20(address reserve, uint256 amountToAave,uint256 amountToCruize) internal {
+        IERC20 token = IERC20(reserve);
+        require(token.transferFrom(msg.sender, address(this), amountToAave));
+        require(token.approve(address(POOL), amountToAave));
+        POOL.deposit(address(token), amountToAave, address(this), 0);
+        require(token.transferFrom(msg.sender,cruize_wallet, amountToCruize));
```

Comment:
why msg.sender i think the token must transfer from Contract it self ?correct me if m wrong